modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -396,7 +396,11 @@ private[smithy4s] class SchemaVisitorJCodec(
         case a: DArray =>
           out.writeArrayStart()
           a.value match {
-            case x: ArraySeq[Document] =>
+            // short-circuiting on empty arrays to avoid the downcast to array of documents
+            // which has proven to be dangerous in Scala 3:
+            // https://github.com/disneystreaming/smithy4s/issues/1158
+            case x: ArraySeq[_] if x.isEmpty =>
```

Comment:
I have no idea whether the compiler optimises this to avoid doing two class-checks : 

```scala 
case a: A if (...) => 
case a: A
```

so I'd just nest the conditional : 

```scala
case x: ArraySeq[_] => 
  if (x.nonEmpty){
     ...
  } 
```

Reply:
4de846993612bf94944ff29b46e2f7532a6c81db